src/test/ui/iterators/array-of-ranges.stderr

```diff
@@ -1,102 +0,0 @@
-error[E0277]: `[std::ops::Range<{integer}>; 1]` is not an iterator
-  --> $DIR/array-of-ranges.rs:2:14
-   |
-LL |     for _ in [0..1] {}
-   |              ^^^^^^ if you meant to iterate between two values, remove the square brackets
```

Comment:
It seems unfortunate to lose this hint, but you can legitimately iterate this with `Item = Range<_>` now, so I don't know how else to help someone who didn't intend that.

Reply:
Can we turn it into a clippy lint?
It might be legal, but it's probably not intended.

Maybe this could lint:
```rust
for _ in [0..1] {}
```

But it could be silenced using:
```rust
for Range<_> in [0..1] {}
```

Reply:
This is now https://github.com/rust-lang/rust-clippy/issues/7125

Reply:
Thanks!